schemas/ufo/glyph.fbs

```diff
@@ -0,0 +1,60 @@
+// we include fontinfo.fbs this because the Guideline table is already defined there
+include "fontinfo.fbs";
+include "plist.fbs";
+
+//namespace FlatFont.Ufo;
+
+/// https://unifiedfontobject.org/versions/ufo3/glyphs/glif/
+table Glyph {
+  name: string (required, key);
+  width: float;
+  height: float;
+  unicodes: [uint];
+  image: Image;
+  lib: Plist.Dict;
+  note: string;
+  anchors: [Anchor];
+  components: [Component];
+  contours: [Contour];
+  guidelines: [Guideline];
+}
+
+table Image {
+  fileName: string;
+  transformation: Transform;
+  color: string;
+}
+
+table Anchor {
+  x: float;
+  y: float;
+  name: string;
+  color: string;
+  identifier: string;
+}
+
+table Component {
+  baseGlyph: string;
+  transformation: Transform;
+  identifier: string;
+}
+
+table Contour {
+  points: [Point];
+  identifier: string;
+}
+
+table Point {
+  x: float;
+  y: float;
+  type: string;
+  smooth: bool;
+  name: string;
+  identifier: string;
+}
+
+struct Transform {
+  values: [float:6];
```

Comment:
Ideally a union with `xScale, xyScale, yxScale, yScale, xOffset, yOffset`. But union can't take struct?

I feel like `Transform` should also live in `base.fbs`. Perhaps in a toplevel `base.fbs` even, not in `ufo/base.fbs` even.

Reply:
why union? We can have Transform strcut use names for each field instead of a fixed-size array, but in defcon/ufoLib2 APIs it is a 6-tuple, not a dictionary.

Yes I can move to base.fbs if you prefer, though it's only used in glyph.

Reply:
Do you prefer like this?

```
namespace FlatFont;

struct Transform {
  xScale: float;
  xyScale: float;
  yxScale: float;
  yScale: float;
  xOffset: float;
  yOffset: float;
}
```

Note that there's been a lot of confusions about these symbolic names for the 2x3 affine fields. In FontTools as well as UFO specification and a lot of the lib that derive from those, the _names_ of `xyScale` and `yxScale` are swapped compared to similar structs found in FreeType and Cairo, while they actually mean the same thing, see: 
https://github.com/fonttools/fonttools/blob/445108f735b22a5ca37f669808d47906d024fe24/Lib/fontTools/ttLib/tables/otData.py#L1633-L1648

Would be best to just avoid the names and use a fixed length array of 6 floats IMO.

Reply:
Even if you use 6 floats the order needs to be specified, so you are not actually *solving* the problem.

I prefer the spelled out.

Also, set the default for `xScale` and `yScale` to 1.0. How about that?

Reply:
Anyway, we can reconsider later.

Reply:
you can set defaults on structs.. should I turn Transform into a `table` to set their defaults?

How shall I call xyScale and yxScale? using fontTools/UFO convention of FreeType/Cairo?

Reply:
(I mean't one cannot set defaults on structs)

Reply:
Yeah let's forget about struct and make it table, with defaults, which would make it stored more optimized.

The naming should be `xyScale` first and `yxScale` after indeed.

Reply:
> The naming should be xyScale first and yxScale after indeed.

funny you say that.. in COLRv1 Affine2x3 we ended up following the FreeType/Cairo naming and have yxScale _before_ xyScale.
(Of course the actual ordering is the same, only the label differ, which adds to the confusion).

Reply:
we call "yxScale" the "y-part of the x basis vector" and xyScale "the x-part of the y basis vector" (following 3blue1brown's definitions) /cc @rsheeter

Reply:
> funny you say that..

Not funny actually that groupthink prevailed...

> we call "yxScale" the "y-part of the x basis vector"

That's English, which is *not* universal. It should be "xyScale", because math / computer code is:
```
x' = xxScale * x + xyScale * y
y' = yxScale * x + yyScale * y
```